pkg/gofr/datasource/opentsdb/opentsdb.go

```diff
@@ -0,0 +1,418 @@
+}
+
+type Config struct {
+
+	// The host of the target opentsdb, is a required non-empty string which is
+	// in the format of ip:port without http:// prefix or a domain.
+	Host string
+
+	// A pointer of http.Transport is used by the opentsdb client.
+	// This value is optional, and if it is not set, client.defaultTransport, which
+	// enables tcp keepalive mode, will be used in the opentsdb client.
+	Transport *http.Transport
+
+	// The maximal number of datapoints which will be inserted into the opentsdb
+	// via one calling of /api/put method.
+	// This value is optional, and if it is not set, client.defaultMaxPutPointsNum
+	// will be used in the opentsdb client.
+	MaxPutPointsNum int
+
+	// The detect delta number of datapoints which will be used in client.Put()
+	// to split a large group of datapoints into small batches.
+	// This value is optional, and if it is not set, client.defaultDetectDeltaNum
+	// will be used in the opentsdb client.
+	DetectDeltaNum int
+
+	// The maximal body content length per /api/put method to insert datapoints
+	// into opentsdb.
+	// This value is optional, and if it is not set, client.defaultMaxPutPointsNum
+	// will be used in the opentsdb client.
+	MaxContentLength int
+}
+
+type Health struct {
+	Status  string         `json:"status,omitempty"`
+	Details map[string]any `json:"details,omitempty"`
+}
+
+// New initializes a new instance of Opentsdb with provided configuration.
+func New(config *Config) *Client {
+	return &Client{config: *config}
+}
+
+func (c *Client) UseLogger(logger interface{}) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+func (c *Client) UseMetrics(metrics interface{}) {
+	if m, ok := metrics.(Metrics); ok {
+		c.metrics = m
+	}
+}
+
+func (c *Client) UseTracer(tracer any) {
+	if tracer, ok := tracer.(trace.Tracer); ok {
+		c.tracer = tracer
+	}
+}
+
+// Connect initializes an HTTP client for OpenTSDB using the provided configuration.
+// If the configuration is invalid or the endpoint is unreachable, an error is logged.
+func (c *Client) Connect() {
+	span := c.addTrace(context.Background(), "Connect")
+
+	status := statusFailed
+
+	var message string
+
+	defer sendOperationStats(c.logger, time.Now(), "Connect", &status, &message, span)
+
+	// Set default values for optional configuration fields.
+	c.initializeClient()
+
+	// Initialize the OpenTSDB client with the given configuration.
+	c.endpoint = fmt.Sprintf("http://%s", c.config.Host)
+
+	c.logger.Logf("Connection Successful")
+
+	status = statusSuccess
+	message = fmt.Sprintf("connected to %s", c.endpoint)
+}
+
+func (c *Client) PutDataPoints(ctx context.Context, datas any, queryParam string, resp any) error {
+	span := c.addTrace(ctx, "Put")
+
+	status := statusFailed
+
+	message := "Put request failed"
+
+	defer sendOperationStats(c.logger, time.Now(), "Put", &status, &message, span)
+
+	putResp, ok := resp.(*PutResponse)
+	if !ok {
+		return errors.New("invalid response type. Must be *PutResponse")
+	}
```

Comment:
This is something I didn't get with the package signature

Everything is passed as any, then the type is checked

What is the reason behind this?

Is it for avoiding coupling in the interface used in containers package?

I would have expected something like this?

```suggestion
func (c *Client) PutDataPoints(ctx context.Context, datas any, queryParam string, resp *PutResponse) error {
	span := c.addTrace(ctx, "Put")

	status := statusFailed

	message := "Put request failed"

	defer sendOperationStats(c.logger, time.Now(), "Put", &status, &message, span)
```

Reply:
If I define actual type in the containers package....goFr package will have dependency added of opentsdb. I want to avoid that, hence as the interface is defined there, I am using any....